src/AutoRest.CSharp/Common/Output/Builders/MultipartSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,265 @@
+                }
+            };
+        }
+
+        private static MethodBodyStatement SerializeValue(VariableReference mulitpartContent, MultipartValueSerialization valueSerialization, ValueExpression valueExpression, StringExpression name)
+        {
+            ValueExpression? contentExpression = null;
+            ValueExpression? fileNameExpression = null;
+            ValueExpression? contentTypeExpression = null;
+            if (valueSerialization.Type.IsFrameworkType && valueSerialization.Type.FrameworkType == typeof(Stream))
+            {
+                contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+                if (valueSerialization.FileName != null)
+                {
+                    fileNameExpression = Literal(valueSerialization.FileName);
+                }
+                else
+                {
+                    fileNameExpression = name.Add(Literal(".wav"));
+                }
+                if (valueSerialization.ContentType != "application/json")
+                {
+                    contentTypeExpression = Literal(valueSerialization.ContentType);
+                }
+            }
+            if (valueSerialization.Type.IsFrameworkType && valueSerialization.Type.FrameworkType == typeof(BinaryData))
+            {
+                contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+                if (valueSerialization.FileName != null)
+                {
+                    fileNameExpression = Literal(valueSerialization.FileName);
+                } else
+                {
+                    fileNameExpression = name.Add(Literal(".wav"));
+                }
+                if (valueSerialization.ContentType != "application/json")
+                {
+                    contentTypeExpression = Literal(valueSerialization.ContentType);
+                }
+            }
+            if (valueSerialization.Type.IsFrameworkType)
+            {
+                contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+            }
+            if (contentExpression != null)
+            {
+                return Configuration.ApiTypes.GetMultipartFormDataRequestContentAddStatment(mulitpartContent, contentExpression, name, fileNameExpression, contentTypeExpression);
+            }
+
+            switch (valueSerialization.Type.Implementation)
+            {
+                case SerializableObjectType serializableObjectType:
+                    contentExpression = BuildValueSerizationExpression(serializableObjectType.Type, valueExpression);
+                    break;
+                case ObjectType:
+                    contentExpression = BuildValueSerizationExpression(valueSerialization.Type, valueExpression);
+                    break;
```

Comment:
would ther e be the second `ObjectType` case? because  `SerializableObjectType` inherits from `ObjectType`.
And we are switching on `valueSerialization.Type`, if this is a `SerializableObjectType`, we use the type. if it is a `ObjectType`, we use the type `valueSerialization.Type`. But they are the same, we should only use one of them.

I think we should only keep the `SerializableObjectType` and let `ObjectType` (if any) error out. Because this is for serialization, a type provider not inheriting `SerializableObjectType` as suggested by its name, should not support any type of serialization.